activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1468,6 +1471,12 @@ def uniq!(name)
     # is passed in) are not instances of the same model that the relation is
     # scoping.
     def excluding(*records)
+      records.map! do |record|
+        case record
+        when QueryMethods then record.select(record.primary_key)
```

Comment:
Is this an exhaustive enough class list? Are there other classes that support `select`?

Reply:
Why not
```ruby
case record
when Relation
  record.ids
  # ...
```
?

Reply:
`QueryMethod` is an implementation detail of ActiveRecord, `Relation` is a public API used everywhere, so I think better to use it.